### PR TITLE
bpo-33911: Fixed deprecation warning in XML-RPC ServerHTMLDoc.

### PR DESCRIPTION
**Pull request by niconoe**
https://bugs.python.org/issue33911

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by vstinner** [comment]
Does your PR change the output? If yes, it should be documented with a new NEWS entry (use the "blurb" tool to add it: https://devguide.python.org/committing/#what-s-new-and-news-entries ).

**Comment by vstinner** [comment]
@1st1: Would you mind to have a look?

File: Lib/xmlrpc/server.py [hunk 1]
```diff
@@ -772,23 +773,13 @@ def docroutine(self, object, name, mod=None,
...
             # confusing to the non-Python user
-            argspec = inspect.formatargspec (
-                    args.args[1:],
-                    args.varargs,
-                    args.varkw,
-                    args.defaults,
-                    annotations=args.annotations,
-                    formatvalue=self.formatvalue
-                )
+            sig.replace(parameters=tuple(sig.parameters.values())[1:])
```
Comment (1st1) [inline Lib/xmlrpc/server.py]: `sig.replace()` returns a new `Signature` object, so you should have written `sig = sig.replace(...)`.  But `inspect.signature` should take care of the bound argument automatically, so you don't need this dance. `if` and `elif` clauses should be replaced with

```python
if callable(object):
    argspec = str(signature(object))
else:
    argspec = '(...)'
```

**Review by 1st1 — DISMISSED** [review]

File: Lib/xmlrpc/server.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/xmlrpc/server.py]: How common a failure of `signature()`? Shouldn't we catch some common exceptions?

Comment (1st1) [reply Lib/xmlrpc/server.py]: `inspect.getfullargspec` uses `inspect.signature` behind the scenes.  So if you want to add some error handling, it should be in a separate PR.

**Comment by niconoe** [comment]
I have made the requested changes; please review again.

@vstinner: AFAIK, the output doesn't change (HTML output is compared in various test methods in `test_docxmlrpc.py`). What I didn't do is to ensure that this existing test coverage is extensive. Should I?

**Review by vstinner — APPROVED** [review]
LGTM.

"./python -W error -m test test_docxmlrpc" now pass, so the PR fixed the issue that I reported.

test_docxmlrpc still pass which means that the HTML is not modified by this change.

**Comment by vstinner** [comment]
@1st1: Would you mind to review again the change?

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**